pkgs/development/python-modules/executor/default.nix

```
@@ -0,0 +1,28 @@
+{ lib, buildPythonPackage, fetchFromGitHub, coloredlogs, property-manager, fasteners }:
+
+buildPythonPackage rec {
+  pname = "executor";
+  version = "21.3";
+
+  src = fetchFromGitHub {
+    owner  = "xolox";
+    repo   = "python-executor";
+    rev    = version;
+    sha256 = "0rc14vjx3d6irfaw0pczzw1pn0xjl7xikv32hc1fvxv2ibnldv5d";
+  };
+
+  propagatedBuildInputs = [ coloredlogs property-manager fasteners ];
+
+  # testing requires sudo, disable
+  # checkInputs = [ virtualenv ];
+  doCheck = false;
```

still preferred to just ignore tests that are impure.
```suggestion
  checkInputs = [ pytest mock virtualenv ];

  checkPhase = ''
    pytest . -k "not option and not retry \
                 and not remote and not ssh \
                 and not foreach and not local_context"
  '';
```